src/ivar.cpp

```diff
@@ -499,7 +499,7 @@ int main(int argc, char* argv[]){
       for (int i = 0; i < nfiles; ++i) {
 	free(files[i]);
       }
-      free(files);
+      delete(files);
```

Comment:
I'm not very familiar with cpp, but iiuc `delete` is a keyword/operator (rather than a function) and since `files` is an array, I think we might want: `delete [] files`(?). I'll make a separate PR just in case it is helpful/quicker to merge.